homeassistant/helpers/update_coordinator.py

```diff
@@ -285,8 +286,12 @@ async def async_config_entry_first_refresh(self) -> None:
         to ensure that multiple retries do not cause log spam.
         """
         if self.config_entry is None:
-            raise ValueError(
-                "This method is only supported for coordinators with a config entry"
+            report(
+                "uses `async_config_entry_first_refresh` without a config entry, "
+                "which is deprecated and will stop working in "
+                "Home Assistant 2025.11",
+                error_if_core=True,
+                error_if_integration=False,
             )
```

Comment:
I'm confused by this deprecation... this never worked, how can something that never worked be deprecated?

Reply:
It did and does work. `self.config_entry` is not used for anything critical.

Reply:
It before raised a ValueError... so it didnt work...

The same logic now it reports a deprecation => Uh wut?

Reply:
> It before raised a ValueError... so it didnt work...

The ValueError was only just added in https://github.com/home-assistant/core/pull/127980

Reply:
Ok, I see. That makes more sense 👍

Reply:
> Ok, I see. That makes more sense 👍

I've also updated the PR description to highlight this.